spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java

```diff
@@ -205,8 +214,14 @@ public void configure(HttpSecurity http) throws Exception {
 	}
 
 	private ClientCredentialsTokenEndpointFilter clientCredentialsTokenEndpointFilter(HttpSecurity http) {
+		
+		FrameworkEndpointHandlerMapping handlerMapping = frameworkEndpointHandlerMapping();
+
 		ClientCredentialsTokenEndpointFilter clientCredentialsTokenEndpointFilter = new ClientCredentialsTokenEndpointFilter(
-				frameworkEndpointHandlerMapping().getServletPath("/oauth/token"));
+				Arrays.asList(handlerMapping.getServletPath("/oauth/token"),
+						handlerMapping.getServletPath("/oauth/token_key"),
```

Comment:
This is not correct. The `ClientCredentialsTokenEndpointFilter` should only match on `/oauth/token`. The default authorization server configuration will setup the authorization rules for `/oauth/token_key` and `/oauth/check_token`.